src/frontend/src/lib/components/core/Listener.svelte

```diff
@@ -9,18 +8,22 @@
 	import { isNetworkIdBitcoin, isNetworkIdICP } from '$lib/utils/network.utils';
 
 	export let token: OptionToken;
-
-	let cmp: ComponentType;
-	$: cmp =
-		isNullish(token) || $authNotSignedIn
-			? NoListener
-			: isNetworkIdICP(token.network.id)
-				? NoListener
-				: isNetworkIdBitcoin(token.network.id)
-					? BitcoinListener
-					: EthListener;
 </script>
 
-<svelte:component this={cmp} {token}>
-	<slot />
-</svelte:component>
+{#if isNullish(token) || $authNotSignedIn}
+	<NoListener>
+		<slot />
+	</NoListener>
```

Comment:
shall we delete `NoListener` at this point? and leave just `<slot />`

Reply:
I was thinking doing that in another PR.